File: app/Models/Presensi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Presensi extends Model
{
    protected $table = 'presensi';
    public $timestamps = false;

    protected $fillable = [
        'kelas',
        'kelas_kuliah_id',
        'pertemuan',
        'mahasiswa_uid',
        'nim',
        'nama',
        'keterangan',
    ];

    public function mahasiswa(): BelongsTo
    {
        return $this->belongsTo(Mahasiswa::class, 'mahasiswa_uid', 'uid');
    }

    public function kelasKuliah(): BelongsTo
    {
        return $this->belongsTo(KelasKuliah::class, 'kelas_kuliah_id', 'id');
    }
}

File: lib/features/presensi/domain/presensi.dart
class Presensi {
  final String uid, nim, nama, keterangan;

  const Presensi({
    required this.uid,
    required this.nim,
    required this.nama,
    required this.keterangan,
  });

  factory Presensi.fromMap(Map<String, dynamic> map) => Presensi(
        uid: map['mahasiswa_uid'] ?? map['uid'] ?? '',
        nim: map['nim'] ?? '',
        nama: map['nama'] ?? '',
        keterangan: map['keterangan'] ?? 'Alpha',
      );
}
